.github/scripts/get_tutorials_stats.py

```diff
@@ -0,0 +1,239 @@
+            "files changed",
+            "file changed",
+            "insertions(+)",
+            "insertion(+)",
+            "deletion(-)",
+            "deletions(-)",
+        ]:
+            x = x.replace(f" {pattern}", "")
+        return x
+
+    title = None
+    rc: List[List[str]] = []
+    for line in lines:
+        # Check for weird entries where subject has double quotes or similar issues
+        if title is None:
+            title = line.split(";", 3)
+        # In the lines with stat, add 0 insertions or 0 deletions to make sure we don't break the table
+        elif "files changed" in line.replace("file changed", "files changed"):
+            stats = do_replace(parse_string(line)).split(",")
+        elif len(line) == 0:
+            rc.append(title + stats)
+            title = None
+        else:
+            rc.append(title + ["0", "0", "0"])
+            title = line.split(";", 3)
+    return rc
+
+def get_file_names(cwd: Optional[str] = None) -> List[Tuple[str, List[Tuple[str, int, int]]]]:
+    lines = run_command('git log --pretty="format:%h" --numstat', cwd=cwd).split("\n")
+    rc = []
+    commit_hash = ""
+    files: List[Tuple[str, int, int]] = []
+    for line in lines:
+        if not line:
+            # Git log uses empty line as separator between commits (except for oneline case)
+            rc.append((commit_hash, files))
+            commit_hash, files = "", []
+        elif not commit_hash:
+            # First line is commit short hash
+            commit_hash = line
+        elif len(line.split("\t")) != 3:
+            # Encountered an empty commit
+            assert(len(files) == 0)
+            rc.append((commit_hash, files))
+            commit_hash = line
+        else:
+            added, deleted, name = line.split("\t")
+            # Special casing for binary files
+            if added == "-":
+                assert deleted == "-"
+                files.append((name, -1, -1))
+            else:
+                files.append((name, int(added), int(deleted)))
+    return rc
+
+def table_exists(table_name: str) -> bool:
+    """
+    Determines whether a table exists. As a side effect, stores the table in
+    a member variable.
+    """
+    try:
+        table = dynamodb.Table(table_name)
+        table.load()
+        exists = True
+        print("Table exists")
+    except ClientError as err:
+        if err.response['Error']['Code'] == 'ResourceNotFoundException':
+            exists = False
+        else:
+            print("Unknown error")
+            pprint(err.response)
+    return exists
+
+def delete_table(table_name: str) -> None:
+    table = dynamodb.Table(table_name)
+    table.delete()
+
+    print(f"Deleting {table.name}...")
+    table.wait_until_not_exists()
+
+def create_table(table_name: str) -> None:
+    """
+    Creates a DynamoDB table.
+
+    :param dyn_resource: Either a Boto3 or DAX resource.
+    :return: The newly created table.
+    """
```

Comment:
This function does not return `None` and does not accept `dyn_resource` as parameter